packages/core/core/src/parcel-v3/Parcel.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import path from 'path';
+import {Worker} from 'worker_threads';
+import * as napi from '@parcel/rust';
+import type {FileSystem} from '@parcel/types';
+
+export type ParcelV3Options = {|
+  threads?: number,
+  nodeWorkers?: number,
+  fs?: FileSystem,
+|};
+
+export class ParcelV3 {
+  _internal: napi.ParcelNapi;
+  #nodeWorkerCount: number;
+
+  constructor({
+    threads = napi.ParcelNapi.defaultThreadCount(),
```

Comment:
Wouldn't it be cleaner to do the defaulting internally?

Reply:
The JavaScript class needs access to this value. I could make it a property of the `ParcelNapi` struct and access it from there - mind if I do that in a follow up PR?

Reply:
Yeah this isn't blocking or anything